src/vs/workbench/contrib/accessibility/browser/accessibleView.ts

```diff
@@ -0,0 +1,160 @@
+import { IModelService } from 'vs/editor/common/services/model';
+import { LinkDetector } from 'vs/editor/contrib/links/browser/links';
+import { localize } from 'vs/nls';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { IContextViewDelegate, IContextViewService } from 'vs/platform/contextview/browser/contextView';
+import { IInstantiationService, createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { SelectionClipboardContributionID } from 'vs/workbench/contrib/codeEditor/browser/selectionClipboard';
+import { getSimpleEditorOptions } from 'vs/workbench/contrib/codeEditor/browser/simpleEditorOptions';
+import { IDisposable } from 'xterm';
+
+interface IAccessibleContentProvider { id: string; provideContent(): string; onClose(): void; options?: IAccessibleViewOptions }
+export const IAccessibleViewService = createDecorator<IAccessibleViewService>('accessibleViewService');
+
+export interface IAccessibleViewService {
+	readonly _serviceBrand: undefined;
+	show(providerId: string): void;
+	registerProvider(provider: IAccessibleContentProvider): void;
+}
+
+export interface IAccessibleViewOptions {
+	isHelpMenu: boolean;
+}
+
+export class AccessibleViewService extends Disposable implements IAccessibleViewService {
+	_serviceBrand: undefined;
+
+	private _providers: Map<string, IAccessibleContentProvider> = new Map();
+	private _editorWidget: CodeEditorWidget;
+	private _editorContainer: HTMLElement;
+
+	constructor(
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+		@IConfigurationService private readonly _configurationService: IConfigurationService,
+		@IModelService private readonly _modelService: IModelService,
+		@IContextViewService private readonly _contextViewService: IContextViewService
+	) {
+		super();
+		this._editorContainer = document.createElement('div');
+		this._editorContainer.classList.add('accessible-view');
+		const codeEditorWidgetOptions: ICodeEditorWidgetOptions = {
+			contributions: EditorExtensionsRegistry.getSomeEditorContributions([LinkDetector.ID, SelectionClipboardContributionID, 'editor.contrib.selectionAnchorController'])
+		};
+		const editorOptions: IEditorConstructionOptions = {
+			...getSimpleEditorOptions(),
+			lineDecorationsWidth: 6,
+			dragAndDrop: true,
+			cursorWidth: 1,
+			wrappingStrategy: 'advanced',
+			wrappingIndent: 'none',
+			padding: { top: 2, bottom: 2 },
+			quickSuggestions: false,
+			renderWhitespace: 'none',
+			dropIntoEditor: { enabled: true },
+			accessibilitySupport: this._configurationService.getValue<'auto' | 'off' | 'on'>('editor.accessibilitySupport'),
+			cursorBlinking: this._configurationService.getValue('terminal.integrated.cursorBlinking'),
+			readOnly: true
+		};
+		this._editorWidget = this._register(this._instantiationService.createInstance(CodeEditorWidget, this._editorContainer, editorOptions, codeEditorWidgetOptions));
```

Comment:
We should lazy load this on `show`. To simplify handling of undefined you could create a class that represents the code editor and its container, so when one exists the other is guaranteed to as well.